docker/setup/terraform/terraform.tfvars

```diff
@@ -0,0 +1,19 @@
+#TODO Using this as an example of the possibility. Given we populate from a docker-compose command,
+# I don't know if this will be super useful for us, vice utilising the Default values found in variables.tf
+# See: https://learn.hashicorp.com/tutorials/terraform/azure-variables 

```

Comment:
I would suggest to add a Bash or PowerShell script to the repo that runs after `terraform apply` which takes the output from `terraform output -json -no-color` and converts it to dotenv format required by `.github.env` or `azure.env`. The encryption key can be stored as a Github Actions secret. The benefit of keeping the dotenv file as an interface is that the rest of the code doesn't have to be changed with the addition of Terraform as it only assumes secrets stored in dotenv.

Encryption can be done via gpg or openssl. Here's an example for openssl:

```sh
# encryption
echo "${password}" | openssl enc -aes-256-cbc -md sha512 -pass stdin -in .env -out .env.enc

# decryption
echo "${password}" | openssl enc -aes-256-cbc -md sha512 -pass stdin -in .env.enc -out .env -d
```

You can see an example of gpg decryption in `makefile`.

Reply:
I am a little unclear on this point. Is the intention to store the output of the terraform apply in the secrets folder as an encrypted .env? So:
Steps:
- terraform apply
- terraform output -json -no-color => azure.env (encrypted) stored in the secrets

Is there anything additional needed for this, that I missed?

Reply:
To clarify a little further, is the intention to use the `terraform output` to populate the values of:
```

RESOURCE_GROUP=${RESOURCE_GROUP_NAME}
LOKOLE_EMAIL_SERVER_APPINSIGHTS_KEY=$(jq -r .properties.outputs.appinsightsKey.value /tmp/deployment.json)
LOKOLE_CLIENT_AZURE_STORAGE_KEY=$(jq -r .properties.outputs.clientBlobsKey.value /tmp/deployment.json)
LOKOLE_CLIENT_AZURE_STORAGE_NAME=$(jq -r .properties.outputs.clientBlobsName.value /tmp/deployment.json)
LOKOLE_CLIENT_AZURE_STORAGE_HOST=
LOKOLE_CLIENT_AZURE_STORAGE_SECURE=True
LOKOLE_EMAIL_SERVER_AZURE_BLOBS_KEY=$(jq -r .properties.outputs.serverBlobsKey.value /tmp/deployment.json)
LOKOLE_EMAIL_SERVER_AZURE_BLOBS_NAME=$(jq -r .properties.outputs.serverBlobsName.value /tmp/deployment.json)
LOKOLE_EMAIL_SERVER_AZURE_BLOBS_HOST=
LOKOLE_EMAIL_SERVER_AZURE_BLOBS_SECURE=True
LOKOLE_EMAIL_SERVER_AZURE_TABLES_KEY=$(jq -r .properties.outputs.serverTablesKey.value /tmp/deployment.json)
LOKOLE_EMAIL_SERVER_AZURE_TABLES_NAME=$(jq -r .properties.outputs.serverTablesName.value /tmp/deployment.json)
LOKOLE_EMAIL_SERVER_AZURE_TABLES_HOST=
LOKOLE_EMAIL_SERVER_AZURE_TABLES_SECURE=True
LOKOLE_EMAIL_SERVER_QUEUES_NAMESPACE=$(jq -r .properties.outputs.serverQueuesName.value /tmp/deployment.json)
LOKOLE_EMAIL_SERVER_QUEUES_SAS_NAME=$(jq -r .properties.outputs.serverQueuesSasName.value /tmp/deployment.json)
LOKOLE_EMAIL_SERVER_QUEUES_SAS_KEY=$(jq -r .properties.outputs.serverQueuesSasKey.value /tmp/deployment.json)
```
which will be stored in the `azure.env` similar to [setup.sh](https://github.com/ascoderu/lokole/blob/master/docker/setup/setup.sh#L81-L99)?